container/appmaster/runit/service/flink/run

```diff
@@ -73,13 +73,37 @@ function add_kerberos_configurations() {
   fi
 }
 
+function add_ha_configurations() {
+  if [[ "${FLINK_HA_ENABLED}" == true ]]; then
+    export FLINK_JAVA_OPTS="$FLINK_JAVA_OPTS -Dhigh-availability=ZOOKEEPER"
```

Comment:
Yeah sure for the first option I missed that. By the way it wont work for every property, there is one that causes problems:
2017-08-02 00:00:20,277 INFO  org.apache.flink.mesos.runtime.clusterframework.MesosApplicationMasterRunner  -     -Dhigh-availability.job.delay="10
2017-08-02 00:00:20,277 INFO  org.apache.flink.mesos.runtime.clusterframework.MesosApplicationMasterRunner  -     s"

The property Dhigh-availability.job.delay="10 s" is not preserved when passed.
In on order to fix this and keep a global variable you need to pass each arg quoted or use eval. The first option is not handy as we have many variables and the second one is not a good choice since it opens security problems if we don't sanitize every input. It gets ugly so I will try HA options in the file. Meanwhile, I am not sure it is going to work since the config.sh script removes quotes and this property needs it. I will test.
Btw I dont think it matters what goes as to the file or as dynamic properties from what I have seen in the code.